pkg/syncer/byokoperator/vcenter/provider.go

```diff
@@ -0,0 +1,103 @@
+package vcenter
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/vmware/govmomi/pbm"
+	cnsvsphere "sigs.k8s.io/vsphere-csi-driver/v3/pkg/common/cns-lib/vsphere"
+	cnsconfig "sigs.k8s.io/vsphere-csi-driver/v3/pkg/common/config"
+)
+
+type Provider interface {
+	DoesProfileSupportEncryption(ctx context.Context, profileID string) (bool, error)
+	ResetVC()
+}
+
+func NewProvider() Provider {
+	return &defaultProvider{}
+}
+
+type defaultProvider struct {
+	client       *cnsvsphere.VirtualCenter
+	reinitialize bool
+	lock         sync.RWMutex
+}
+
+func (p *defaultProvider) DoesProfileSupportEncryption(ctx context.Context, profileID string) (bool, error) {
+	pbmClient, err := p.getPbmClient(ctx)
+	if err != nil {
+		return false, err
+	}
+	return pbmClient.SupportsEncryption(ctx, profileID)
+}
+
+func (p *defaultProvider) ResetVC() {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	p.client = nil
+	p.reinitialize = true
+}
+
+func (p *defaultProvider) getVcClient(ctx context.Context) (*cnsvsphere.VirtualCenter, error) {
```

Comment:
Can we use existing util file we have to get VC Client?
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/util.go#L51-L68